File: 0027-remove-element/0027-remove-element.py
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
       # we use two piinters one at the start and one at the end
       # we check if the first pointer points to val if exchange it with the val of the last pointer 

        left, right = 0, len(nums)-1

        while left <= right:
            if nums [left] == val:
                nums[left], nums[right] = nums[right], nums[left]
                right -= 1
                del nums[right+1]
            else:
                left += 1
        

File: 0027-remove-element/0027-remove-element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int right = std::size(nums);
        int left = 0;

        while ( right > left) {
            if(nums[left] == val) {
                nums[left] = nums[right - 1];
                right--;
            }
            else{
                left++;
            }
        }
        return right;
    }
};